### PR TITLE
[READY]Blacklists tools, part exchangers, and beakers from the deep fryer. Allows them to be unanchored, and deconstructed

### PR DESCRIPTION
**Pull request by silicons**
Goofball, fix your shit. Yes, this is necessary.
By the way where the hell is the maint panel open icon?

### CODE REVIEW

**Comment by Iamgoofball** [comment]
The tools and beakers I understand but that should be handled with ..()

Eating the nuke disk is the #1 asked thing when I pitch the fryer. No.

**Comment by francinum** [comment]
Deep-fried nuke disks are a time-honored tradition!

**Comment by B00tz** [comment]
Fun police coming through.

**Comment by kevinz000** [comment]
brain/nuke disk blacklist removed.

**Comment by PKPenguin321** [comment]
it works fine as is

**Comment by kevinz000** [comment]
@PKPenguin321 I hope you mean minus being undeconstructable, unmovable, and unable to be filled without frying your beaker. frankly at this state i'd use them for a higher level of walls seeing as they can't be damaged with anything but a gun by human players.

**Comment by xxalpha** [comment]
Don't blacklist plutonium cores, you anti fun. Let people enjoy their rad snacks.

**Comment by kevinz000** [comment]
@xxalpha Ok fine I'll take that out too but the tool/rped/beakers are staying. reeeeEEEEEE.

File: code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm [hunk 1]
```diff
@@ -18,7 +18,7 @@ insert ascii eagle on american flag background here
 	container_type = OPENCONTAINER
 	var/obj/item/frying = null	//What's being fried RIGHT NOW?
 	var/cook_time = 0
-
+	var/list/blacklisted_items
```
Comment (coiax) [inline code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm]: I think this can be a static list

Comment (RemieRichards) [inline code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm]: 1. it must be ```/static``` I demand it
2. making it ```/static``` means you can move ```... = typecacheof(...)``` into the definition here, which is better as it'll only run once vs n-times

File: code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm [hunk 2]
```diff
@@ -46,8 +47,16 @@ insert ascii eagle on american flag background here
...
 		return
+	if(default_unfasten_wrench(user, I))
+		return
+	else if(exchange_parts(user, I))
+		return
+	else if(default_deconstruction_screwdriver(user, "fryer_off", "fryer_off" ,I))	//where's the open maint panel icon?!
+		return
 	else
-		if(user.drop_item() && !frying)
+		if(I in blacklisted_items)
```
Comment (RemieRichards) [inline code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm]: this is not how you typecache.

```is_type_in_typecache(I, blacklisted_items)```

**Comment by PKPenguin321** [comment]
Why blacklist beakers? Just just add a check before adding reagents then return after it adds the reagents

File: code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm [hunk 3]
```diff
@@ -18,6 +18,7 @@ insert ascii eagle on american flag background here
 	container_type = OPENCONTAINER
 	var/obj/item/frying = null	//What's being fried RIGHT NOW?
 	var/cook_time = 0
+	var/static/list/blacklisted_items = typecacheof(list(/obj/item/weapon/screwdriver, /obj/item/weapon/crowbar, /obj/item/weapon/wrench, /obj/item/weapon/wirecutters, /obj/item/device/multitool, /obj/item/weapon/weldingtool, /obj/item/weapon/reagent_containers/glass, /obj/item/weapon/storage/part_replacer))
```
Comment (coiax) [inline code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm]: WHY DID YOU NOT SPLIT THIS LINE UP REEEEEEEEEEEEEe

**Pull request merged**